backend/internal/service/handler/users/user.go

```diff
@@ -125,6 +125,44 @@ func (h *Handler) GetUserProfile(c *fiber.Ctx) error {
 	return c.Status(fiber.StatusOK).JSON(profile)
 }
 
+
+func (h *Handler) UpdateUserProfilePicture(c *fiber.Ctx) error {
+	id := c.Params("id")
+
+	exists, err := h.userRepository.UserExists(c.Context(), id)
+	if err != nil {
+		print(err.Error())
+		return err
+	}
+
+	if !exists {
+		return errs.NotFound("User", "userID", id)
+	}
+
+	userUUID, err := uuid.Parse(id)
+	if err != nil {
+		print(err.Error())
+		return err
+	}
+
+	var requestBody struct {
```

Comment:
fail fast by parsing the body at the top of this func. if the body isn't valid, your user exists check is for nought.